### ISSUE TITLE
Why .js files are split into parts?

### ISSUE THREAD

**Issue by webpolis** [gwatts/jquery.sparkline#92]
Hi

You have header.js, base.js... all code split into parts.. why is that? It makes very difficult to include this package using bower or grunt tools, or do minification tasks over the source.

Please advise.

Thanks.

**Comment by soullivaneuh** [comment]
I just asked the same question in #70 

There is a makefile, but it is not convenient for direct implementation...

**Comment by gwatts** [comment]
they're split into parts to make it easier to develop.

Building it is very simple; I can't imagine it'd be hard to have grunt do the same thing the Makefile does now - If you're motivated to come up with a suitable grunt script then feel free - I haven't spent anytime looking into either bower or grunt.

I'm not interested in checking in a built copy of the plugin into git though

**Comment by soullivaneuh** [comment]
And why not a git repo for builts ?

Indeed, in some project, we can add a hooks to launch the makefile, like this in composer.json for Symfony2:

``` json
"scripts": {
    "post-install-cmd": [
        "cd src/Acme/MainBundle/Resources/public/lib/sparkline && make all",
    ],
    "post-update-cmd": [
        "cd src/Acme/MainBundle/Resources/public/lib/sparkline && make all",
    ]
```

But install uglify-js and compile in each update in a production server, not really good... :/

**Comment by gwatts** [comment]
You don't need to build it at all if you just download the release off the project web site - the git repo is for development

**Comment by soullivaneuh** [comment]
Bower is a dependencies manager, thanks to it, we can get latest minor version that is good think, because we don't always check new version of all js libraries ourself.

Sorry for my english. ;)

**Comment by gwatts** [comment]
All well and good, but if you want the bleeding edge you'll have to build it, as you would with almost any other github project (including jQuery itself afaik)

**Comment by soullivaneuh** [comment]
Source repo of jQuery: https://github.com/jquery/jquery
Builts repo of jQuery: https://github.com/components/jquery

But not the same user, maybe we have to contact us in order to propose your buildted package...

**Comment by gwatts** [comment]
If you want to figure out what steps are needed to have a build of the plugin hosted in the components project, then I'll gladly take a pull request

**Comment by kapusta** [comment]
Fwiw, I just submitted a PR that Grunt-ifies the build process.

**Comment by mariusGundersen** [comment]
Bower does not use the bleeding edge, it uses tagged releases. It is very common to include a built version in the repo for bower. It would be great if you could include it too.

**Comment by arcseldon** [comment]
Bower does not use the bleeding edge, it uses tagged releases. It is very common to include a built version in the repo for bower. It would be great if you could include it too. +1

**Comment by okendoken** [comment]
Totally agree with @mariusGundersen & @arcseldon 

Guys, please include built version for bower users.

**Comment by seglo** [comment]
+1

**Comment by follesoe** [comment]
Had same issue as many of you. But - as a tip you can use bower to install single files from any HTTP URL: `bower install --save http://omnipotent.net/jquery.sparkline/2.1.2/jquery.sparkline.js` will pull in sparkline for you.

**Comment by neverfox** [comment]
That's true @follesoe, and it's the solution I'm having to use now. But it's less than ideal because it cannot respond to new releases with semver.

**Comment by gwatts** [comment]
The latest release is now also at http://omnipotent.net/jquery.sparkline/latest/jquery.sparkline.js fwiw

**Comment by neverfox** [comment]
Good to know. Thanks.

**Comment by tony** [comment]
@gwatts : Any updates here? jQuery's latest version switched to modularize their plugin, have you considered that approach?

> The latest release is now also at http://omnipotent.net/jquery.sparkline/latest/jquery.sparkline.js 
> 
> bower install --save http://omnipotent.net/jquery.sparkline/2.1.2/jquery.sparkline.js
> 
> I'm not interested in checking in a built copy of the plugin into git though

I feel you, keeping refs to built versions of the plugin isn't philosophically pure or efficient. Other plugins are breaking this best practice, they've come to the consensus that having a built version available is friendly and consistent to developers.

In the here and now, people would opt to have a built version in your git they can grab via bower.

In the long term, future versions of bower will address this issue.

**Comment by JGailor** [comment]
+1 for built version of this.

**Comment by vladmiller** [comment]
+1 for this feature
It's not hard to do and a lot of people begging for this, why just don't simply add precompiled version to the repo?

**Comment by geminiyellow** [comment]
+1 
i want bower support.

**Comment by A-owl** [comment]
+1 bower support.

**Comment by estliberitas** [comment]
+2 for bower support.

@gwatts How can I help to force things?

**Comment by anthonyvialleton** [comment]
+1

**Comment by ccravens** [comment]
+1 for bower support
